### PR TITLE
Feat/navigation popover

### PR DESCRIPTION
**Pull request by fabio-tedesco**
Extend bal-navigation-popover behaviours and add two popover in stories with content examples for both the mobile and the desktop vp

### CODE REVIEW

File: packages/components/src/components/bal-navigation/bal-navigation.tsx [hunk 1]
```diff
@@ -43,6 +47,21 @@ export class Navigation implements ComponentInterface {
         this.selectedMainValue = ''
       }
     }
+
+    if (isPlatform('touch')) {
+      this.mainNavMobile = this.el.querySelector('.bal-nav__mainmobile') as HTMLElement
```
Comment (hirsch88) [inline packages/components/src/components/bal-navigation/bal-navigation.tsx]: - [x] please create get method to trap the elements / avoid duplicated code

File: packages/components/src/components/bal-navigation/bal-navigation.tsx [hunk 2]
```diff
@@ -95,6 +118,38 @@ export class Navigation implements ComponentInterface {
...
+      await this.allowScrollingBody()
+    }
+  }
+
+  /**
+   * Allows the scrolling on the body element
+   */
+  @Method()
+  async allowScrollingBody() {
+    this.body.style.position = 'static'
```
Comment (hirsch88) [inline packages/components/src/components/bal-navigation/bal-navigation.tsx]: - [x] Could we create a global util function to do so.

File: packages/components/src/components/bal-navigation/bal-navigation.tsx [hunk 3]
```diff
@@ -95,6 +118,38 @@ export class Navigation implements ComponentInterface {
...
+    this.body.style.width = 'auto'
+    this.body.style.overflowY = 'visible'
+  }
+
+  /**
+   * Blocks the scrolling on the body element
+   */
+  @Method()
+  async blockScrollingBody() {
+    this.body.style.position = 'fixed'
```
Comment (hirsch88) [inline packages/components/src/components/bal-navigation/bal-navigation.tsx]: - [x]  Could we create a global util function to do so.

File: packages/components/src/components/bal-popover/bal-popover.tsx [hunk 4]
```diff
@@ -224,12 +235,44 @@ export class Popover {
...
+      await this.allowScrollingBody()
+    }
+  }
+
+  /**
+   * Allows the scrolling on the body element
+   */
+  @Method()
+  async allowScrollingBody() {
+    this.body.style.position = 'static'
```
Comment (hirsch88) [inline packages/components/src/components/bal-popover/bal-popover.tsx]: - [x] use the global util function / duplicated code

File: packages/components/src/components/bal-popover/bal-popover.tsx [hunk 5]
```diff
@@ -224,12 +235,44 @@ export class Popover {
...
+    this.body.style.width = 'auto'
+    this.body.style.overflowY = 'visible'
+  }
+
+  /**
+   * Blocks the scrolling on the body element
+   */
+  @Method()
+  async blockScrollingBody() {
+    this.body.style.position = 'fixed'
```
Comment (hirsch88) [inline packages/components/src/components/bal-popover/bal-popover.tsx]: - [x] use the global util function / duplicated code

File: packages/components/src/components/bal-popover/bal-popover.tsx [hunk 6]
```diff
@@ -224,12 +235,44 @@ export class Popover {
...
+    this.body.style.overflowY = 'hidden'
+  }
+
   private updatePopper() {
     this.popperInstance.update()
 
     // to trigger a popper rerender
-    window.scrollTo(window.scrollX, window.scrollY - 1)
-    window.scrollTo(window.scrollX, window.scrollY + 1)
+    /*window.scrollTo(window.scrollX, window.scrollY - 1)
```
Comment (hirsch88) [inline packages/components/src/components/bal-popover/bal-popover.tsx]: - [ ] what about this commented code ?

**Review by hirsch88 — CHANGES_REQUESTED** [review]

File: packages/components/src/components/bal-popover/bal-popover.tsx (hunk 6 above)
Comment (fabio-tedesco) [reply packages/components/src/components/bal-popover/bal-popover.tsx]: deleted

File: packages/components/src/components/bal-navigation/bal-navigation.tsx [hunk 7]
```diff
@@ -36,13 +41,32 @@ export class Navigation implements ComponentInterface {
...
+        this.isMainBodyOpen &&
+        this.mainNavMobile &&
+        this.mainNavFootMobile
+      ) {
+        this.isMainBodyOpen = false
+      }
+    }
+  }
+
+  getMobileNavElement(selector: string) {
```
Comment (hirsch88) [inline packages/components/src/components/bal-navigation/bal-navigation.tsx]: - [x] no no I meant for each selector create a getter method like get mainMobileElement

File: packages/components/src/components/bal-navigation/bal-navigation.tsx [hunk 8]
```diff
@@ -203,11 +231,15 @@ export class Navigation implements ComponentInterface {
...
               slot="burger"
-              color="light"
+              color={this.isMainBodyOpen ? 'primary' : 'light'}
               square={true}
               icon={this.isMainBodyOpen ? 'close' : 'menu-bars'}
-              onClick={() => (this.isMainBodyOpen = !this.isMainBodyOpen)}
-              inverted={this.isMainBodyOpen}
+              onClick={async () => {
+                this.isMainBodyOpen = !this.isMainBodyOpen
+                //await this.toggleScrollingBody()
```
Comment (hirsch88) [inline packages/components/src/components/bal-navigation/bal-navigation.tsx]: - [x] please remove comments

File: packages/components/src/components/bal-navigation/bal-navigation.tsx [hunk 9]
```diff
@@ -203,11 +231,15 @@ export class Navigation implements ComponentInterface {
...
+                console.log('this.isMainBodyOpen BURGER ICON touch', this.isMainBodyOpen)
```
Comment (hirsch88) [inline packages/components/src/components/bal-navigation/bal-navigation.tsx]: - [x] Remove console.logs

**Review by hirsch88 — CHANGES_REQUESTED** [review]

**Review by hirsch88 — APPROVED** [review]

**Pull request merged**